x-pack/plugins/enterprise_search/public/applications/__mocks__/wait_for_promises.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/**
+ * If you created an immediately resolved promise, you can use this to wait for it to resolve without a reference.
+ *
+ * http.get.mockReturnValueOnce(Promise.resolve({ analyticsUnavailable: true }));
+ * AnalyticsLogic.actions.loadQueryData('some-query');
+ * 
+ * await waitForPromises();
+        
+ * expect(AnalyticsLogic.actions.onAnalyticsUnavailable).toHaveBeenCalled();
```

Comment:
I think I'd prefer a more generic and less specific example for understanding, but maybe that's just my individual preference as a dev 🤔 

```suggestion
 * http.get.mockReturnValueOnce(Promise.resolve({ someData: true }));
 * SomeLogic.actions.loadData();
 *
 * await waitForPromises();
 *
 * expect(http.get).toHaveBeenCalled();
 * expect(SomeLogic.actions.onLoadData).toHaveBeenCalled(); // Any action expected as a result of data being fetched
```